lookout/core/lib.py

```diff
@@ -87,38 +91,104 @@ def files_by_language(files: Iterable[File]) -> Dict[str, Dict[str, File]]:
+        if exclude_compiled_pattern and exclude_compiled_pattern.search(filepath):
+            continue
+        passed.append(filepath)
+    return passed
+
+
+def filter_files_by_line_length(filepaths: Iterable[str], line_length_limit: int) -> Iterable[str]:
+    """
+    Filter files that have lines longer than `line_length_limit`.
+
+    :param filepaths: paths to the files to filter.
+    :param line_length_limit: maximum line length to accept a file.
+    :return: files passed through the line-length filter.
+    """
+    line_passed = []
+    for filepath in filepaths:
+        with open(filepath, "rb") as f:
+            content = f.read()
+        if len(max(content.splitlines(), key=len, default=b"")) <= line_length_limit:
+            line_passed.append(filepath)
+    return line_passed
+
+
+def parse_files(filepaths: Iterable[str], line_length_limit: int, overall_size_limit: int,
+                client: BblfshClient, language: str, random_state: int = 7,
+                progress_tracker: Callable = lambda x: x, log: Optional[logging.Logger] = None
+                ) -> Iterable[File]:
+    """
+    Parse files with Babelfish.
+
+    If a file has lines longer than `line_length_limit`, it is skipped. If the summed size of \
+    parsed files exceeds `overall_size_limit` the rest of the files is skipped. Files paths are \
+    filtered with `filter_filepaths()`. The order in which the files are parsed is random - \
+    and hence different from `filepaths`.
+
+    :param filepaths: paths to the files to filter.
     :param line_length_limit: maximum line length to accept a file.
     :param overall_size_limit: maximum cumulative files size in bytes. \
                                The files are discarded after reaching this limit.
+    :param client: Babelfish client. Babelfish server should be started accordingly.
+    :param language: Language to consider. Will discard the other languages.
     :param random_state: random generator state for shuffling the files.
+    :param progress_tracker: Optional progress metric whenn iterating over the input files.
     :param log: logger to use to report the number of excluded files.
     :return: files passed through the filter and the number of files which were excluded.
     """
-    line_passed = []
-    for key, file in files.items():
-        if len(max(file.content.splitlines(), key=len, default=b"")) <= line_length_limit:
-            line_passed.append(key)
-    if log is not None:
-        log.debug("excluded %d/%d files by max line length %d",
-                  len(files) - len(line_passed), len(files), line_length_limit)
-    line_passed.sort()
     random.seed(random_state)
-    line_passed = random.sample(line_passed, k=len(line_passed))
-    size = 0
+    filepaths_filtered = filter_filepaths(filepaths)
+    filepaths_filtered = random.sample(filepaths_filtered, k=len(filepaths_filtered))
+    files_filtered_by_line_length = filter_files_by_line_length(filepaths_filtered,
```

Comment:
You should swap here: first filter by line length and then shuffle.

Reply:
How about renaming `filter_filepaths` to `filter_files_by_path`? The two functions will be uniformly named then.